apps/src/lib/kits/maker/boards/circuitPlayground/CircuitPlaygroundBoard.js

```diff
@@ -92,33 +92,61 @@ export default class CircuitPlaygroundBoard extends EventEmitter {
    */
   connectToFirmware() {
     return new Promise((resolve, reject) => {
-      const name = this.port_ ? this.port_.comName : undefined;
-      const serialPort = CircuitPlaygroundBoard.openSerialPort(name);
-      const playground = CircuitPlaygroundBoard.makePlaygroundTransport(
-        serialPort
-      );
-      const board = new five.Board({io: playground, repl: false, debug: false});
-      board.once('ready', () => {
-        this.serialPort_ = serialPort;
-        this.logWithFirehose(
-          'serial-port-set',
-          JSON.stringify({serialPort, name})
+      // A port that exists, but doesn't have a comName is a Web Serial port
+      if (this.port_ && !this.port_.comName) {
```

Comment:
You mentioned that right now this is support just for Adafruit boards. For other boards, like micro:bit, will this way  of determining if it's a Web Serial port still be true? Just wondering if it is how we want to determine this property, and will be used for other boards, if we want this as a util like `isWebSerialPort` or something. That change can also come later when we go to implement! 

Also, just curious cause I was wondering this before when looking at CP code, any idea why we use an `_` for `this.port_` ?

Reply:
Once we expand this to micro:bit, we'll add corresponding changes to the MicroBitBoard.js file: https://github.com/code-dot-org/code-dot-org/blob/42d98deb14212889d335bf3817c4393d10c19d9d/apps/src/lib/kits/maker/boards/microBit/MicroBitBoard.js#L97
How we approach the port will be the same, so we will still want to determine which type of port we have. I definitely think we'll want to break out the logic into `isWebSerialPort`. Makes sense to put it in the a boardUtils file somewhere. Or even `portScanning`, maybe.

As far as the underscore? I don't know. I know that prepending an underscore in Javascript is a way to notate private variables/functions, but I'm not sure what the underscore at the end means.

Reply:
Added a helper function!